lib/share_progress/button.rb

```diff
@@ -32,8 +47,42 @@ def endpoint(method=nil)
       end
 
       def filter_keys(params, acceptable)
+        return params if params.nil?
         params.select{ |key, _| acceptable.include? key }
       end
+
+      def new_from_fields(fields)
+        new(fields['id'],
+            fields['page_url'],
+            fields['button_template'],
+            fields['page_title'],
+            fields['share_button_html'],
+            fields['is_active'])
+      end
+
+      def optional_keys
+        [:page_title, :auto_fill, :variations, :advanced_options]
+      end
+
+      def advanced_options_keys
+        [:automatic_traffic_routing, :buttons_optimize_actions, :custom_params, :prompt]
+      end
+    end
+
+    def initialize(id, page_url, button_template, page_title, share_button_html, is_active)
+      self.id = id
+      self.page_url = page_url
+      self.button_template = button_template
+      self.page_title = page_title
+      self.share_button_html = share_button_html
+      self.is_active = is_active
```

Comment:
Yeah that's exactly how I was imagining usage, and what I was wondering if makes sense in terms of mimicking the AR API so much.